ipq40xx: add support for Netgear SRR60/SRS60 and RBR50/RBS50

DavideFioravanti opened the pull request:
These patches add support for Netgear SRR60/SRS60 and RBR50/RBS50 and mmc based devices in general.

The patch [base-files: add minimal mmc support](https://github.com/openwrt/openwrt/commit/878f465cb4c228bf6a8775150009a6c4a3b22061) can be used to standardize the upgrade procedure for mmc devices like the Zyxel NBG6817

~~For what concerns the [ipq40xx: add support for Netgear SRR60/SRS60 and RBR50/RBS50](https://github.com/openwrt/openwrt/commit/e83244a78f63cf85c63ed900ae40fcbefe260881) there is currently a "bug" in the IPQ40XX switch driver which prevents the port 1 to correctly detect the cable connection [(Read more)](https://forum.openwrt.org/t/netgear-orbi-pro-support-booting-from-mmc/14208/209).
I don't know if a fix for this problem will be ready before the introduction of the DSA driver for ipq40xx, so in the meantime I set `qcom,poll_required = <0>;` for the WAN port. The only downside of this is that the WAN port always results as connected even if not.~~
~~When this [PR](https://github.com/openwrt/openwrt/pull/3498) will be accepted, the wan port problem will be solved.~~
Merged. WAN port is working now.

DavideFioravanti committed:
ipq40xx: enable CONFIG_CMDLINE_PARTITION and CONFIG_LEDS_TLC591XX

CONFIG_CMDLINE_PARTITION:
Some devices with mmc like the Netgear Orbi Pro SRS60 or Netgear Orbi
RBR50 needs to hardcode the partitions layout in the cmdline boot
correctly

CONFIG_LEDS_TLC591XX:
This is needed for the led driver found in the Netgear Orbi Pro SRS60

Signed-off-by: Davide Fioravanti <pantanastyle@gmail.com>
Signed-off-by: Robert Marko <robimarko@gmail.com>
Signed-off-by: Shang Jia <jiash416@gmail.com>

DavideFioravanti committed:
base-files: add minimal mmc support

Added minimal mmc support for sysupgrade and some helper functions:

 - find_mmc_part: Look for a given partition name. Returns the
	coresponding partition path
 - caldata_extract_mmc: Look for a given partition name and then
	extracts the calibration data
 - mmc_get_mac_binary: Returns the mac address from a given partition
	name and offset

Signed-off-by: Davide Fioravanti <pantanastyle@gmail.com>
Signed-off-by: Robert Marko <robimarko@gmail.com>

DavideFioravanti committed:
ipq40xx: add support for Netgear SRR60/SRS60 and RBR50/RBS50

The Netgear SRS60 and SRR60 (sold together as SRK60) are two almost
identical AC3000 routers. The SRR60 has one port labeled as wan while
the SRS60 not. The RBR50 and RBS50 (sold together as RBK50) have a
different external shape but they have an USB 2.0 port on the back.

This patch has been tested only on SRS60 and RBR50, but should works
on SRR60 and RBS50.

Hardware
--------
SoC:   Qualcomm IPQ4019 (717 MHz, 4 cores 4 threads)
RAM:   512MB DDR3
FLASH: 4GB EMMC
ETH:
  - 3x 10/100/1000 Mbps Ethernet
  - 1x 10/100/1000 Mbps Ethernet (WAN)
WIFI:
  - 2.4GHz: 1x IPQ4019 (2x2:2)
  - 5GHz:   1x IPQ4019 (2x2:2)
  - 5GHz:   1x QCA9884 (4x4:4)
  - 6 internal antennas
BTN:
  - 1x Reset button
  - 1x Sync button
  - 1x ON/OFF button
LEDS:
  - 8 leds controlled by TLC59208F (they can be switched on/off
    independendently but the color can by changed by GPIOs)
  - 1x Red led (Power)
  - 1x Green led (Power)
UART:
  - 115200-8-N-1

Everything works correctly.

Installation
------------
These routers have a dual partition system. However this firmware works
only on boot partition 1 and the OEM web interface will always flash on
the partition currently not booted.

The following steps will use the SRS60 firmware, but you have to chose
the right firmware for your router.

There are 2 ways to install Openwrt the first time:

1) Using NMRPflash
 1. Download nmrpflash (https://github.com/jclehner/nmrpflash)
 2. Put the openwrt-ipq40xx-generic-netgear_srs60-squashfs-factory.img
	file in the same folder of the nmrpflash executable
 3. Connect your pc to the router using the port near the power button.
 4. Run "nmrpflash -i XXX -f openwrt-ipq40xx-generic-netgear_srs60-squashfs-factory.img".
	Replace XXX with your network interface (can be identified by
	running "nmrpflash -L")
 5. Power on the router and wait for the flash to complete. After about
	a minute the router should boot directly to Openwrt. If nothing
	happens try to reboot the router. If you have problems flashing
	try to set "10.164.183.253" as your computer IP address

2) Without NMRPflash
The OEM web interface will always flash on the partition currently not
booted, so to flash OpenWrt for the first time you have to switch to
boot partition 2 and then flash the factory image directly from the OEM
web interface.

To switch on partition 2 you have to enable telnet first:
 1. Go to http://192.168.1.250/debug.htm and check "Enable Telnet".
 2. Connect through telent ("telnet 192.168.1.250") and login using
	admin/password.

	To read the current boot_part:
		artmtd -r boot_part

	To write the new boot_part:
		artmtd -w boot_part 02

	Then reboot the router and then check again the current booted
	partition

Now that you are on boot partition 2 you can flash the factory Openwrt
image directly from the OEM web interface.

Restore OEM Firmware
--------------------
 1. Download the stock firmware from official netgear support.
 2. Follow the nmrpflash procedure like above, using the official
	Netgear firmware (for example SRS60-V2.2.1.210.img)

        nmrpflash -i XXX -f SRS60-V2.2.1.210.img

Notes
-----
1) You can check and edit the boot partition in the Uboot shell using
	the UART connection.
	"boot_partition_show" shows the current boot partition
	"boot_partition_set 1" sets the current boot partition to 1

2) Router mac addresses:

   LAN XX:XX:XX:XX:XX:69
   WAN XX:XX:XX:XX:XX:6a
   WIFI 2G XX:XX:XX:XX:XX:69
   WIFI 5G XX:XX:XX:XX:XX:6b
   WIFI 5G (2nd) XX:XX:XX:XX:XX:6c

   LABEL XX:XX:XX:XX:XX:69

Signed-off-by: Davide Fioravanti <pantanastyle@gmail.com>
Signed-off-by: Robert Marko <robimarko@gmail.com>

DavideFioravanti commented:
@robimarko As I wrote in the forum, I think that you should be added as co-developer of this patch giving the fact that lot of my work is based on yours, but you didn't respond. I would add you, but I don't know if it's the right thing to do without your consensus... :thinking:

robimarko commented:
Hi, I was not really following the thread after I had to return the SRK60 pair I had for testing.
Feel free adding my SoB:
Signed-off-by: Robert Marko <robimarko@gmail.com>

About the switch driver, yeah that one is a mess of old Qualcomm GPL sources that were updated to work with new kernels.
There is some work to switch to QCA8k based DSA one but it will take a lot of work and time.

I see that you figured out the cmdline partition trick, I tried that also but forgot that it needs to be enabled first.
I am glad to see this thing alive and kicking.

adschm commented on package/base-files/files/lib/upgrade/mmc.sh:
```diff
@@ -0,0 +1,78 @@
+#!/bin/sh
```
That's a library file, remove the shebang and the execute bit (if there is one).

DavideFioravanti replied:
Removed, thanks!

adschm commented on package/base-files/files/lib/upgrade/mmc.sh:
```diff
@@ -0,0 +1,78 @@
+
+mmc_do_upgrade() {
+
+    command -v losetup >/dev/null || {
```
broken indent here. I would also remove the empty line before.

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:
```diff
@@ -0,0 +1,327 @@
+		compatible = "gpio-leds";
+
+		power-green {
+			label = "orbi:power:green";
```
Typically, the color is in the middle, so I would except
orbi:green:power
here.

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:
```diff
@@ -0,0 +1,327 @@
+			panic-indicator;
+		};
+
+		green: green {
```
These names/labels are inconsistent. I'd use something like

led_status_green: status_green {

or

led_status_green: status-green {

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:
```diff
@@ -0,0 +1,327 @@
+		reg = <0x27>;
+		
+		led0@0 {
+			label = "orbi:led0:rgb";
```
See above for color:function order. I have never dealt with rgb leds though, so I don't know what's the best practice there.

DavideFioravanti replied:
The color of these leds is set by the GPIOS defined before... so these "rgb" leds definitions are used just to turn on or off the leds

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbr50.dts:
```diff
@@ -0,0 +1,35 @@
+	};
+
+	soc {
+
```
remove empty line.

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbr50.dts:
```diff
@@ -0,0 +1,35 @@
+#include "qcom-ipq4019-orbi.dtsi"
+
+/ {
+    model = "Netgear RBR50";
```
indent.

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts:
```diff
@@ -0,0 +1,35 @@
+#include "qcom-ipq4019-orbi.dtsi"
+
+/ {
+    model = "Netgear RBS50";
```
indent.

DavideFioravanti replied:
Fixed, thanks!

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts:
```diff
@@ -0,0 +1,35 @@
+	};
+
+	soc {
+
```
remove empty line

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-srr60.dts:
```diff
@@ -0,0 +1,12 @@
+#include "qcom-ipq4019-orbi.dtsi"
+
+/ {
+    model = "Netgear SRR60";
```
indent; and for the next file again.

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/image/Makefile:
```diff
@@ -617,6 +617,63 @@ define Device/netgear_ex6150v2
+	SOC := qcom-ipq4019
+	DEVICE_VENDOR := Netgear
+	BLOCKSIZE := 128k
+	IMAGE/factory.img := append-kernel | pad-offset $$$$(BLOCKSIZE) 64 | append-uImage-fakehdr filesystem | pad-to $$$$(KERNEL_SIZE) | append-rootfs | pad-rootfs | netgear-dni
```
Please wrap lines.

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/image/Makefile:
```diff
@@ -617,6 +617,63 @@ define Device/netgear_ex6150v2
+	KERNEL_SIZE := 3932160
+	ROOTFS_SIZE := 32243712
+	IMAGE_SIZE := 36175872
+	DEVICE_VENDOR := Netgear
```
You already have the vendor in netgear_orbi

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/image/Makefile:
```diff
@@ -617,6 +617,63 @@ define Device/netgear_ex6150v2
+	KERNEL_SIZE := 3932160
+	ROOTFS_SIZE := 32243712
+	IMAGE_SIZE := 36175872
+	DEVICE_VENDOR := Netgear
```
already in netgear_orbi

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/image/Makefile:
```diff
@@ -617,6 +617,63 @@ define Device/netgear_ex6150v2
+define Device/netgear_rbr50
+	$(call Device/netgear_rbx50)
+	NETGEAR_BOARD_ID := RBR50
+	DEVICE_MODEL := RBR50
```
I'd put DEVICE_MODEL before NETGEAR_BOARD_ID here and for the others below.

DavideFioravanti replied:
Fixed, thanks!

DavideFioravanti commented:
@adschm I am really sorry for the errors in the commit. :disappointed: I forgot to check it with checkpatch.pl...

DavideFioravanti commented:
Rebased to current master and updated the wan port configuration to make it working correctly when https://github.com/openwrt/openwrt/pull/3498 will be accepted.
I'll remove the draft status of this PR when the aforementioned PR will be upstreamed

robimarko commented:
@DavideFioravanti I hope it will get merged soon, so I can stop rebasing multiple PR-s that depend on it now.
I also have some EDMA fixes as well as fiber support queued as soon as it's merged.

DavideFioravanti commented:
I hope so too... And above all, I hope that will be merged before the 20.x branch :crossed_fingers:

robimarko commented:
@DavideFioravanti Can you add your Tested-by tag to my PHY PR?
https://github.com/openwrt/openwrt/pull/3498

That would highly increase the chances of it getting merged.

DavideFioravanti commented:
Rebased and tested again cherry-picking these commits: https://github.com/openwrt/openwrt/pull/3498. It's working correctly.

DavideFioravanti commented:
Rebased and tested after the merge of this PR https://github.com/openwrt/openwrt/pull/3498

theoribeiro commented:
@DavideFioravanti and @robimarko 
Commented on the forum post but not sure if you're following the topic. 
I'm testing this PR on my RBS50 (v2 if I'm not mistaken) and getting the following output on UART:
```
Format: Log Type - Time(microsec) - Message - Optional Info
Log Type: B - Since Boot(Power On Reset),  D - Delta,  S - Statistic
S - QC_IMAGE_VERSION_STRING=BOOT.BF.3.1.1-00123
S - IMAGE_VARIANT_STRING=DAACANAZA
S - OEM_IMAGE_VERSION_STRING=CRM
S - Boot Config, 0x00000025
S - Reset status Config, 0x00000000
S - Core 0 Frequency, 0 MHz
B -       261 - PBL, Start
B -      1338 - bootable_media_detect_entry, Start
B -      2608 - bootable_media_detect_success, Start
B -      2622 - elf_loader_entry, Start
B -      4028 - auth_hash_seg_entry, Start
B -      6177 - auth_hash_seg_exit, Start
B -     74171 - elf_segs_hash_verify_entry, Start
B -    197602 - PBL, End
B -    197626 - SBL1, Start
B -    291052 - pm_device_init, Start
D -         7 - pm_device_init, Delta
B -    292552 - boot_flash_init, Start
D -     95234 - boot_flash_init, Delta
B -    392087 - boot_config_data_table_init, Start
D -     13941 - boot_config_data_table_init, Delta - (419 Bytes)
B -    408780 - clock_init, Start
D -      7562 - clock_init, Delta
B -    419783 - CDT version:2,Platform ID:8,Major ID:1,Minor ID:0,Subtype:1
B -    423195 - sbl1_ddr_set_params, Start
B -    428292 - cpr_init, Start
D -         2 - cpr_init, Delta
B -    432674 - Pre_DDR_clock_init, Start
D -         4 - Pre_DDR_clock_init, Delta
D -     13173 - sbl1_ddr_set_params, Delta
B -    445986 - pm_driver_init, Start
D -         2 - pm_driver_init, Delta
B -    516547 - sbl1_wait_for_ddr_training, Start
D -        29 - sbl1_wait_for_ddr_training, Delta
B -    531946 - Image Load, Start
D -    145927 - QSEE Image Loaded, Delta - (274048 Bytes)
B -    678300 - Image Load, Start
D -      2114 - SEC Image Loaded, Delta - (2048 Bytes)
B -    688506 - Image Load, Start
D -    179397 - APPSBL Image Loaded, Delta - (424967 Bytes)
B -    868297 - QSEE Execution, Start
D -        56 - QSEE Execution, Delta
B -    874471 - SBL1, End
D -    678955 - SBL1, Delta
S - Flash Throughput, 2040 KB/s  (701834 Bytes,  343936 us)
S - DDR Frequency, 672 MHz


U-Boot 2012.07-25708-gfbe2156 [Chaos Calmer 15.05.1,r35193] (Oct 25 2018 - 15:15:28)

U-boot dni1 V0.5 for DNI HW ID: 29765913; NOR flash 0MB; NAND flash 128MB; RAM 256MB
smem ram ptable found: ver: 1 len: 3
DRAM:  256 MiB
machid : 0x8010001
NAND:  ID = 9500f1ef
Vendor = ef
Device = f1
ONFI device found
SF NAND unsupported id:ff:ff:ff:ffSF: Unsupported manufacturer ff
ipq_spi: SPI Flash not found (bus/cs/speed/mode) = (0/0/48000000/0)
128 MiB
MMC:   qca_mmc: 0
*** Warning - bad CRC, using default environment

PCI0 Link Intialized
In:    serial
Out:   serial
Err:   serial
Valid chip addresses: 27 48 4A
machid: 8010001
flash_type: 2
Net:    131072 bytes read: OK
MAC0 addr:bc:a5:11:b2:91:c3
PHY ID1: 0x4d
PHY ID2: 0xd0b1
ipq40xx_ess_sw_init done
eth0
Hit any key to stop autoboot:  0
eth0 PHY0 Down Speed :10 Half duplex
eth0 PHY1 Down Speed :10 Half duplex
eth0 PHY2 Down Speed :10 Half duplex
eth0 PHY3 Down Speed :10 Half duplex
eth0 PHY4 Down Speed :10 Half duplex

 nmrp server is stopped or failed !
Loading DNI firmware for checking...
Loading firmware 1 ...

NAND read: device 0 offset 0x3280000, size 0x20000
 131072 bytes read: OK

NAND read: device 0 offset 0x3280000, size 0x2a0000
 2752512 bytes read: OK
dniimg_len is 2752512 (aligned to 2752512)

NAND read: device 0 offset 0x3280000, size 0x2a0000
 2752512 bytes read: OK
if iminfo 0x84000000; then echo kernel checksum OK !;if iminfo 0x8429FFC0; then echo rootfs checksum OK !;boot_partition_set 1;true;else echo rootfs checksum error !!!;fw_recovery;false;fi;else echo kernel checksum error !!!;fw_recovery;false;fi;
## Checking Image at 84000000 ...
   FIT image found
   FIT description: ARM OpenWrt FIT (Flattened Image Tree)
    Image 0 (kernel@1)
     Description:  ARM OpenWrt Linux-5.4.86
     Type:         Kernel Image
     Compression:  uncompressed
     Data Start:   0x840000e4
     Data Size:    2702640 Bytes = 2.6 MiB
     Architecture: ARM
     OS:           Linux
     Load Address: 0x80208000
     Entry Point:  0x80208000
     Hash algo:    crc32
     Hash value:   7d831d33
     Hash algo:    sha1
     Hash value:   3ebef3573af8efd71157277ceebe052796af6393
    Image 1 (fdt@1)
     Description:  ARM OpenWrt netgear_rbs50 device tree blob
     Type:         Flat Device Tree
     Compression:  uncompressed
     Data Start:   0x84293f4c
     Data Size:    17925 Bytes = 17.5 KiB
     Architecture: ARM
     Hash algo:    crc32
     Hash value:   de7c73b1
     Hash algo:    sha1
     Hash value:   fdd0712c2ba0efa2ef02faa2d841a16611f668bc
    Default Configuration: 'config@1'
    Configuration 0 (config@1)
     Description:  OpenWrt
     Kernel:       kernel@1
     FDT:          fdt@1
## Checking hash(es) for FIT Image at 84000000 ...
   Hash(es) for Image 0 (kernel@1): crc32+ sha1+
   Hash(es) for Image 1 (fdt@1): crc32+ sha1+
kernel checksum OK !

## Checking Image at 8429ffc0 ...
   Legacy image found
   Image Name:   OpenWrt fake filesystem
   Image Type:   ARM Linux Filesystem Image (uncompressed)
   Data Size:    0 Bytes = 0 Bytes
   Load Address: 00000000
   Entry Point:  00000000
   Verifying Checksum ... OK
rootfs checksum OK !
Burn BOOT PARTITION (= 1) into boarddata2 block
 131072 bytes read: OK
Erasing: off d80000, size 20000
Erasing at 0xd80000 -- 100% complete. Cleanmarker written at 0xd80000.
OK
Writing: from RAM addr 871cf9c8, to NAND off d80000, size 20000
 131072 bytes written: OK
Alive-timer 11
Done.
## Booting kernel from FIT Image at 84000000 ...
ERROR: can't get kernel image!
```

I can't help much with understanding what is going on but I'm glad to try things on this device and provide outputs if it can help. Let me know if there's anything you need me to try! 
Thanks for your efforts so far!

DavideFioravanti commented:
> Commented on the forum post but not sure if you're following the topic.
I read your comment but I didn't answer because I am very busy at the moment. Sorry.

> I'm testing this PR on my RBS50 (v2 if I'm not mistaken)

I wasn't aware of the v2 model. In the following days I will update this commit to specify the v1 version in the name.

However reading your bootlog it seems to not be compatible with this commit because your router uses NAND flash instead of the MMC of the v1. Unfortunately I can't do anything about it.

adschm commented on target/linux/ipq40xx/base-files/etc/board.d/02_network:
```diff
@@ -104,6 +104,14 @@ ipq40xx_setup_interfaces()
+	netgear,rbs50|\
+	netgear,srr60|\
+	netgear,srs60)
+		ucidef_set_interfaces_lan_wan "eth0" "eth1"
```
Just using ucidef_set_interface_wan should be enough, the lan should be picked up from the switch command.

DavideFioravanti replied:
I tested with your suggested changes and it's working correclty. Thanks

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:
```diff
@@ -0,0 +1,326 @@
+		compatible = "gpio-leds";
+
+		power_green {
+			label = "orbi:green:power";
```
Please remove the "orbi:" prefix for all leds, we just use color:function now ...

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:
```diff
@@ -0,0 +1,326 @@
+		power_red {
+			label = "orbi:red:power";
+			gpios = <&tlmm 64 GPIO_ACTIVE_HIGH>;
+			default-state = "off";
```
default-state=off is default and can be dropped.

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:
```diff
@@ -0,0 +1,326 @@
+		reg = <0x27>;
+
+		led0@0 {
+			label = "orbi:rgb:led0";
```
same here, drop orbi: prefix for all leds.

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-orbi.dtsi:
```diff
@@ -0,0 +1,326 @@
+		compatible = "ti,tlc59108";
+		reg = <0x27>;
+
+		led0@0 {
```
Just led@0, led@1 etc.?

DavideFioravanti replied:
Those leds create a ring on the top of the router and they are not labeled at all. They change color depending on which GPIO in the previous led section is HIGH.
![image](https://user-images.githubusercontent.com/2470733/107159323-953d6280-698f-11eb-8fed-24ca34bba36e.png)
I don't know how to name them.

adschm commented on target/linux/ipq40xx/image/Makefile:
```diff
@@ -650,6 +650,62 @@ define Device/netgear_ex6150v2
+define Device/netgear_orbi
+	$(call Device/DniImage)
+	SOC := qcom-ipq4019
+	DEVICE_VENDOR := Netgear
```
Please use all-caps "NETGEAR"

DavideFioravanti replied:
Fixed, thanks!

adschm commented on target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-rbs50.dts:
```diff
@@ -0,0 +1,30 @@
+#include "qcom-ipq4019-orbi.dtsi"
+
+/ {
+	model = "Netgear RBS50";
```
Please use all-caps NETGEAR for model in all DTS files for consistency with existing devices.

DavideFioravanti replied:
Fixed, thanks!

DavideFioravanti commented:
Fixed, rebased and tested again.
I also added the V1 version for RBR50 and RBS50 to avoid any confusion:
![image](https://user-images.githubusercontent.com/2470733/107159563-34168e80-6991-11eb-9264-600e34c96840.png)
According to the Netgear website currently the SRS60 and SRR60 don't have a V2 version.

Bixilon commented:
@DavideFioravanti Thanks, but before anybody reviews this, you need to solve the merge conflicts

DavideFioravanti commented:
Sorry for the delay. The conflicts were caused by new devices added.
I rebased, fixed the conflicts and tested again.

Bixilon commented:
Any updates?

Flole998 commented:
Can this get reviewed and merged please?

Flole998 commented:
@adschm Could you please review this one again? It's ready to be merged IMO.

chunkeey commented:
Ok, let's give this a try. 

I changed various things (and added 5.10). It would be great if the cmdline partition string could be shorter. Merged!

ArtemiisFowl commented:
I have a netgear ex8000, it should be hardware identical to obri rbr50/rbs50 devices. What's my todo to make this device compatible with openwrt?

Flole998 commented:
Try to run the initramfs and see what happens.

J0hnMatrix commented:
Hello,
I have RBK50v2 (without USB port), is there something I can provide for it's integration in this repo?

Flole998 commented:
Come up with a PR for it and it'll be merged eventually at some point....
